lib/plugins/aws/package/compile/events/websockets/lib/deployment.js

```diff
@@ -9,8 +10,12 @@ module.exports = {
       const routeLogicalId = this.provider.naming.getWebsocketsRouteLogicalId(event.route);
       return routeLogicalId;
     });
+    const cfTemplateHash = crypto
+      .createHash('sha256')
+      .update(JSON.stringify(this.serverless.service.provider.compiledCloudFormationTemplate))
```

Comment:
Maybe let's filter the template to take only resources which names contain `Websockets` token.

Otherwise we may land in an issue, where in all cases we have different hash, as for some resource names `serverless.instanceId` is used, which technically is a timestamp (different with every deployment).

Reply:
Added.

I ended up choosing `WebsocketsApi` as a token for filtering the template. This is the logical id of `AWS::ApiGatewayV2::Api` resource type. This way, all resources that depend on it or have a ref to it are added to part of the template to be hashed. 

The examples of included resource types are routes, permissions, authorizers, integrations, and stage. Lambdas, lambda versions, log groups, and roles were skipped.

I also skipped the outputs section of the template. Only filtering through resources.

I hope this should be good enough.